scripts/sentry-xcodebuild.sh

```diff
@@ -57,63 +63,122 @@ fi
 
 case $COMMAND in
 "build")
+    RUN_ANALYZE=false
     RUN_BUILD=true
     RUN_BUILD_FOR_TESTING=false
     RUN_TEST_WITHOUT_BUILDING=false
+    RUN_TEST_WITH_TSAN=false
     ;;
 "build-for-testing")
+    RUN_ANALYZE=false
     RUN_BUILD=false
     RUN_BUILD_FOR_TESTING=true
     RUN_TEST_WITHOUT_BUILDING=false
+    RUN_TEST_WITH_TSAN=false
     ;;
 "test-without-building")
+    RUN_ANALYZE=false
     RUN_BUILD=false
     RUN_BUILD_FOR_TESTING=false
     RUN_TEST_WITHOUT_BUILDING=true
+    RUN_TEST_WITH_TSAN=false
+    ;;
+"analyze")
+    RUN_ANALYZE=true
+    RUN_BUILD=false
+    RUN_BUILD_FOR_TESTING=false
+    RUN_TEST_WITHOUT_BUILDING=false
+    RUN_TEST_WITH_TSAN=false
+    ;;
+"tsan")
+    RUN_ANALYZE=false
+    RUN_BUILD=false
+    RUN_BUILD_FOR_TESTING=false
+    RUN_TEST_WITHOUT_BUILDING=false
+    RUN_TEST_WITH_TSAN=true
     ;;
 *)
+    RUN_ANALYZE=false
     RUN_BUILD=false
     RUN_BUILD_FOR_TESTING=true
     RUN_TEST_WITHOUT_BUILDING=true
+    RUN_TEST_WITH_TSAN=false
     ;;
 esac
 
-if [ "$CI" == true ]; then
+if [ "${CI-false}" == true ]; then
     RUBY_ENV_ARGS=""
 else
     RUBY_ENV_ARGS="rbenv exec bundle exec"
 fi
 
 if [ $RUN_BUILD == true ]; then
-    set -o pipefail && env NSUnbufferedIO=YES xcodebuild \
+    set -o pipefail && env NSUnbufferedIO=YES CODE_SIGN_IDENTITY=\"\" CODE_SIGNING_REQUIRED=NO xcodebuild \
```

Comment:
I am not sure if we can add these env values here, as we are also building signed versions of the sample apps.